roxctl/common/io/io.go

```diff
@@ -34,13 +37,53 @@ func (i *ioImpl) ErrOut() io.Writer {
 	return i.errOut
 }
 
+func getOutputFile(filePath string) *os.File {
+	// This function opens a file in write (append) mode for output or error writing.
+	// A nil output means that the specified path is either empty, or
+	// that something prevented the file from being opened in write mode.
+	if len(filePath) == 0 {
+		return nil
+	}
+	fileInfo, err := os.Stat(filePath)
+	if os.IsNotExist(err) {
+		// If the output file does not exist, try to create it along with the path to it.
+		dirPath := filepath.Dir(filePath)
+		err := os.MkdirAll(dirPath, 0755)
+		if err != nil {
+			// Could not create directory to store the output file
+			// Revert to default output
+			return nil
+		}
+		file, err := os.OpenFile(filePath, os.O_CREATE|os.O_RDWR|os.O_APPEND, 0600)
+		if err != nil {
+			return nil
+		}
+		return file
+	}
+	if fileInfo.Mode()&0200 != 0 {
```

Comment:
What happens here if `err != nil && !os.IsNotExist(err)`? I assume `fileInfo` will be nil and `fileInfo.Mode()` will panic; I'd rather return in l.47 when `err != nil && !os.IsNotExist(err)`.

Reply:
Good catch.